diff --git a/src/modules/mavlink/mavlink_receiver.cpp b/src/modules/mavlink/mavlink_receiver.cpp
index 48d0b57..40e7802 100644
--- a/src/modules/mavlink/mavlink_receiver.cpp
+++ b/src/modules/mavlink/mavlink_receiver.cpp
@@ -1811,7 +1811,8 @@ MavlinkReceiver::handle_message_battery_status(mavlink_message_t *msg)
 	battery_status.remaining = (float)battery_mavlink.battery_remaining / 100.0f;
 	battery_status.discharged_mah = (float)battery_mavlink.current_consumed;
 	battery_status.cell_count = cell_count;
-	battery_status.temperature = (float)battery_mavlink.temperature;
+	battery_status.temperature = battery_mavlink.temperature == INT16_MAX ? NAN : (float)battery_mavlink.temperature * 0.01f;
+	battery_status.time_remaining_s = battery_mavlink.time_remaining > 0 ? (float)battery_mavlink.time_remaining : NAN;
 	battery_status.connected = true;
 
 	// Set the battery warning based on remaining charge.
